br/pkg/lightning/backend/external/byte_reader.go

```diff
@@ -81,7 +83,14 @@ func (r *byteReader) readNBytes(n int) (*[]byte, error) {
 	for readLen < n {
 		r.cloneSlices()
 		err := r.reload()
-		if err != nil {
+		switch err {
+		case nil:
+		case io.EOF:
+			if readLen > 0 && readLen < n {
```

Comment:
`readLen < n` seems to be always true.

Reply:
https://github.com/pingcap/tidb/pull/45913/commits/6e4a4f7f0b8002389ff28acab11039f5a1b108f6